pkg/topology/full/full.go

```diff
@@ -130,6 +133,50 @@ func (d *Driver) ChunkPeer(addr swarm.Address) (peerAddr swarm.Address, err erro
 	return swarm.Address{}, topology.ErrNotFound
 }
 
+// SyncPeer returns a peer to which we would like to sync an arbitrary
+// chunk address. Returns the closest peer in relation to the chunk.
+func (d *driver) SyncPeer(addr swarm.Address) (swarm.Address, error) {
+	connectedPeers := d.p2pService.Peers()
+	if len(connectedPeers) == 0 {
+		return swarm.Address{}, topology.ErrNotFound
+	}
+
+	overlays := make([]swarm.Address, len(connectedPeers))
+	for i, v := range connectedPeers {
+		overlays[i] = v.Address
+	}
+
+	return closestPeer(addr, d.base, overlays)
+}
+
+func closestPeer(addr, self swarm.Address, peers []swarm.Address) (swarm.Address, error) {
+	// start checking closest from _self_
+	closest := self
+	for _, peer := range peers {
+		dcmp, err := swarm.DistanceCmp(addr.Bytes(), closest.Bytes(), peer.Bytes())
+		if err != nil {
+			return swarm.Address{}, err
+		}
+		switch dcmp {
+		case 0:
+			// do nothing
+		case -1:
+			// current peer is closer
+			closest = peer
```

Comment:
@janos do i need to shadow `peer` on `L156` as a result?

Reply:
I do not think that it is needed.